domain/application/service/unit.go

```diff
@@ -226,6 +231,49 @@ func (s *Service) UpdateCAASUnit(ctx context.Context, unitName coreunit.Name, pa
 	return nil
 }
 
+// SetReportedUnitAgentVersion sets the reported agent version for the
+// supplied unit name. Reported agent version is the version that the agent
+// binary on this unit has reported it is running.
+//
+// The following errors are possible:
+// - [coreerrors.NotValid] if the reportedVersion is not valid.
+// - [coreerrors.NotSupported] if the architecture is not supported.
+// - [applicationerrors.UnitNotFound] - when the unit does not exist.
+// - [applicationerrors.UnitIsDead] - when the unit is dead.
+func (s *Service) SetReportedUnitAgentVersion(ctx context.Context, unitName coreunit.Name, reportedVersion agentbinary.Version) error {
+	if err := reportedVersion.Validate(); err != nil {
+		return errors.Errorf("reported agent version %v is not valid: %w", reportedVersion, err)
+	}
+
+	unitLife, err := s.st.GetUnitLife(ctx, unitName)
```

Comment:
This needs to be done in a txn. 

We have this method already in state
`checkUnitNotDead(ctx context.Context, tx *sqlair.TX, ident unitUUID) error`

We need to use the above method in the state `SetRunningAgentBinaryVersion` method.
We need to change this service method `SetReportedUnitAgentVersion` to take a unit UUID instead of a name and do the unit UUID lookup in the facade. The end goal is that unit agent will use the UUID when making api calls, not the name, we can need to design the service api to take that into account.